framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -4774,6 +4819,112 @@ VkResult VulkanReplayConsumerBase::OverrideCreateDebugUtilsMessengerEXT(
+{
+    gfxrecon::decode::VulkanResourceAllocator* resource_allocator =
+        reinterpret_cast<gfxrecon::decode::VulkanResourceAllocator*>(allocator_class);
+    gfxrecon::decode::VulkanResourceAllocator::MemoryData data;
+    VkResult res = resource_allocator->AllocateMemoryDirect(allocate_info, allocation_callbacks, memory, &data);
+    if (res == VK_SUCCESS && resource_id != nullptr)
+    {
+        *resource_id = static_cast<uintptr_t>(data);
+    }
+    return res;
+}
+
+void FreeDeviceMemoryResource(void*                        allocator_class,
+                              VkDeviceMemory               memory,
+                              const VkAllocationCallbacks* allocation_callbacks,
+                              uintptr_t                    resource_id)
+{
+    gfxrecon::decode::VulkanResourceAllocator* resource_allocator =
+        reinterpret_cast<gfxrecon::decode::VulkanResourceAllocator*>(allocator_class);
+    gfxrecon::decode::VulkanResourceAllocator::MemoryData data =
+        static_cast<gfxrecon::decode::VulkanResourceAllocator::MemoryData>(resource_id);
+    resource_allocator->FreeMemoryDirect(memory, allocation_callbacks, data);
+}
+
+VkResult BindImageResourceToDeviceMemoryResource(void*                  allocator_class,
+                                                 VkImage                image,
+                                                 VkDeviceMemory         memory,
+                                                 VkDeviceSize           memory_offset,
+                                                 uintptr_t              image_resource_id,
+                                                 uintptr_t              memory_resource_id,
+                                                 VkMemoryPropertyFlags* bind_memory_properties)
+{
+    gfxrecon::decode::VulkanResourceAllocator* resource_allocator =
+        reinterpret_cast<gfxrecon::decode::VulkanResourceAllocator*>(allocator_class);
+    gfxrecon::decode::VulkanResourceAllocator::ResourceData resource_data =
+        static_cast<gfxrecon::decode::VulkanResourceAllocator::ResourceData>(image_resource_id);
+    gfxrecon::decode::VulkanResourceAllocator::MemoryData memory_data =
+        static_cast<gfxrecon::decode::VulkanResourceAllocator::MemoryData>(memory_resource_id);
+    return resource_allocator->BindImageMemoryDirect(
+        image, memory, memory_offset, resource_data, memory_data, bind_memory_properties);
+}
+
+void LogErrorMessage(const char* message, ...)
+{
+    va_list args;
+    va_start(args, message);
+    GFXRECON_LOG_ERROR(message, args);
+    va_end(args);
+}
+
+void LogWarningMessage(const char* message, ...)
+{
+    va_list args;
+    va_start(args, message);
+    GFXRECON_LOG_WARNING(message, args);
+    va_end(args);
+}
+
+void LogInfoMessage(const char* message, ...)
+{
+    va_list args;
+    va_start(args, message);
+    GFXRECON_LOG_INFO(message, args);
+    va_end(args);
+}
```

Comment:
GFXRECON_LOG is in utils\loggings.h. I feel gfxrecon_util.lib is the lowest level library. compatibility could just use gfxrecon_util.lib.

Reply:
@locke-lunarg : This is going to be compiled into non-GFXR code (i.e. the generated "toCpp" output).  My original design did not intend for that source to have to link back into GFXR source since they may not have it on the generating system.  Is this important to maintain?  If so, it will likely require more changes.